src/rust/engine/VERSION

```diff
@@ -0,0 +1 @@
+../../python/pants/VERSION
```

Comment:
@thejcannon: FYI: this is a symlink to allow the native client to embed this resource in its binary via `include_str`. Possibly relevant to your bootstrapping changes.

Reply:
We might have to discuss this. The file might move to only exist in the sandbox

Reply:
Just thinking through an option here, which convinces me that there's a plausible path forward:

- when building Rust outside pants, we could have a build script that runs the same process to compute the right version file ✅ 
- when building Rust via pants (i.e. #19817), the Rust build can depend on the file in the normal pants way (same as the PEX), and the build script can read that instead of running the generation process itself  ✅ 

I.e. a build script that does something like `read_file() if file_exists() else generate_version()` to output a file or set an env var.

Reply:
I agree. I think the only thing is to make sure we keep the on-disk version "fresh" from option 1. E.g. make sure that it's _always_ generated for manual builds. Otherwise it might get generated once, and then not again, and reflects some old value over time